### PR TITLE
Typesense index deploy

### PR DESCRIPTION
**Pull request by mayhem**
This PR adds atomic typesense index rotation, stable MBID mapping calculation, stable year lookup calculation, with each being re-created atomically every 24 hours. All services remain fully in operation while indexes are being updated.

Most of the code in this PR is cleanup, except for the typesense index rotation code, which is actually new.

### CODE REVIEW

File: listenbrainz/mbid_mapping/docker/crontab [hunk 1]
```diff
@@ -1,2 +1,2 @@
-# Create the recording pairs and mapping each monday and thursday
-0 4 * * 1,5 listenbrainz /usr/local/bin/python /code/mapper/manage.py create-all >> /code/mapper/lb-cron.log 2>&1
+# Create the typesend index once a day
```
Comment (alastair) [inline listenbrainz/mbid_mapping/docker/crontab:1]: typesense

**Comment by mayhem** [comment]
The usual heisenbugs hanging about....

File: listenbrainz/mbid_mapping/mapping/typesense_index.py [hunk 2]
```diff
@@ -80,11 +117,15 @@ def build_index():
...
-                    print(i)
+                if i and i % 1000000 == 0:
+                    log("typesense index: Indexed %d rows" % i)
 
             if documents:
-                client.collections[COLLECTION_NAME].documents.import_(documents)
+                client.collections[collection_name].documents.import_(documents)
+
+    log("typesense index: indexing complete. waiting for background tasks to finish.")
+    sys.stdout.flush()
```
Comment (alastair) [inline listenbrainz/mbid_mapping/mapping/typesense_index.py:130]: probably don't need the flush if you're already doing it in `log`

File: listenbrainz/mbid_mapping/mapping/typesense_index.py [hunk 3]
```diff
@@ -80,11 +117,15 @@ def build_index():
...
-                if i and i % 100000 == 0:
-                    print(i)
+                if i and i % 1000000 == 0:
+                    log("typesense index: Indexed %d rows" % i)
 
             if documents:
-                client.collections[COLLECTION_NAME].documents.import_(documents)
+                client.collections[collection_name].documents.import_(documents)
+
+    log("typesense index: indexing complete. waiting for background tasks to finish.")
```
Comment (alastair) [inline listenbrainz/mbid_mapping/mapping/typesense_index.py:129]: just out of interest, what background tasks are happening at this point?

**Review by alastair — CHANGES_REQUESTED** [review]
Looks mostly OK. I think we can remove some duplication in the Dockerfile

File: listenbrainz/mbid_mapping/mapping/typesense_index.py (hunk 3 above)
Comment (mayhem) [reply listenbrainz/mbid_mapping/mapping/typesense_index.py:129]: Dunno, its not specified in the docs. But if I terminate right away, I get nasty sounding error messages.

**Review by alastair — APPROVED** [review]
thank you!

**Pull request merged**